server/src/main/java/org/opensearch/indices/IndicesModule.java

```diff
@@ -278,6 +279,7 @@ protected void configure() {
         bind(RetentionLeaseSyncAction.class).asEagerSingleton();
         bind(RetentionLeaseBackgroundSyncAction.class).asEagerSingleton();
         bind(RetentionLeaseSyncer.class).asEagerSingleton();
+        bind(SegmentReplicationCheckpointPublisher.class).asEagerSingleton();
```

Comment:
I think we should gate this with the feature flag if we can.  The constructor that requires this component wouldn't be invoked with the flag off.

Reply:
In IndicesModule we don't have access to IndexSettings or IndexMetaData through which we can get value of feature flag, so it is not possible to gate this with feature flag in here. Need to figure out some different way to bind this conditionally

Reply:
@Rishikesh1159 looks like you're talking about the index setting? @mch2 ' s comment refers to the [seg-rep feature flag](https://github.com/opensearch-project/OpenSearch/blob/ee7b731975d9044d3239ee3a4fe97f50d45af3e5/server/src/main/java/org/opensearch/common/util/FeatureFlags.java#L22)

Reply:
>  to actually see if seg-rep is on or not we have to see IndexSettings or IndexMetaData

That is correct - seg-rep on an index is controlled by an index setting. However, while the feature is in development, we are requiring that all integrations with seg-rep related classes be gated (directly or indirectly) by the feature flag. This allows us to iteratively build out the full functionality without risking existing, stable code paths. By binding `SegmentReplicationCheckpointPublisher` here without a check, we risk affecting other code paths in case that component has a bug.

Hopefully that makes this a little clearer?

Reply:
Got it, yes I have much clear understanding now. Thanks @kartg